### ISSUE TITLE
Encountered error in KeySink

### ISSUE THREAD

**Issue by BartSte** [kmonad/kmonad#453]
When running the master branch from the command line I get the following error:

- Encountered error in KeySink: Cannot translate to windows keycode: Missing254
- Encountered error in <emitter_proc>: Cannot translate to windows keycode: Missing254

When running it with logging set to debug, it first fails with the same error code. When I run it the second time, however, it works!

[sys.txt](https://github.com/kmonad/kmonad/files/7797543/sys.txt)
[my_config.txt](https://github.com/kmonad/kmonad/files/7797548/my_config.txt)

**Comment by Seinsbyte** [comment]
Same issue here but running it a second time does not help

**Comment by Seinsbyte** [comment]
Hey I did some testing today and this issue seems to be caused by the frequency of underlines you have in your code (for passthrough). So having a .kbd file with lots of passthroughs makes it so that I almost always get a Missing254 error, and having a file with 0 passthroughs seems to work every time. My "solution" atm is to just use passthroughs sparingly and try running the code over and over again until it manages to load without the missing 254 error (you could also not use passthroughs at all but that seems to stop key repeat from working).

**Comment by BartSte** [comment]
Alright, thanks for your help!

**Comment by Haprog** [comment]
I have the same issue (built from master) trying to run with basic home row mods in config. But running it multiple times doesn't seem to help. It always fails with the same error and kmonad doesn't seem to do anything.

With debug log level this is the output:
```
Opening KeySink                                                                                                         
Initializing Windows key sink                                                                                           
Opening KeySource                                                                                                       
Registering low-level Windows keyboard hook                                                                             
Launching process: emitter_proc                                                                                         
                                                                                                                        
--------------------------------------------------------------------------------                                        
Received event: Press <missing254>                                                                                      
Running hooks                                                                                                           
Registering untimed hook: 1                                                                                             
Emitting: Press <missing254>                                                                                            
                                                                                                                        
--------------------------------------------------------------------------------                                        
Received event: Press <missing254>                                                                                      
Encountered error in KeySink: Cannot translate to windows keycode: Missing254                                           
Running hooks                                                                                                           
Encountered error in <emitter_proc>: Cannot translate to windows keycode: Missing254                                    
Registering untimed hook: 2                                                                                             
Closing process: emitter_proc                                                                                           
Registering untimed hook: 3                                                                                             
Closing KeySource                                                                                                       
Unregistering low-level Windows keyboard hook
```

**Comment by Haprog** [comment]
No such problem if running with the prebuilt `kmonad-0.4.1-win.exe` binary from GH release. That seems to work fine without errors using the same config.

**Comment by montanacodes** [comment]
Still having this issue in .4.1 windows exe. It prevents automatically running kmonad at startup.

**Comment by montanacodes** [comment]
If anyone runs into this snag my temporary fix has been to find and replace all my underscores on alt layers with "lctrl". I don't seem to get the errors and it has caused no unintended harm that I've noticed.

**Comment by tupini07** [comment]
I get the same error on Windows 11, but interesting enough everything works fine if I set the log level to `debug` 🤔

**Comment by Kabouik** [comment]
I am getting the same error and none of the above tips worked for me, with ad69e1b compiled for aarch64. When I run `kmonad myconf.kbd`, I get the KeySynk error as soon as I press a key. 

My configuration is a rather simple one at the moment:

```lisp
(defcfg
  input  (device-file "/dev/input/by-path/platform-c17a000.i2c-event-kbd")
  output (uinput-sink "My KMonad output"
    ;; To understand the importance of the following line, see the section on
    ;; Compose-key sequences at the near-bottom of this file.
    "/usr/bin/sleep 1 && /usr/bin/setxkbmap -option compose:ralt")
  cmp-seq capslock    ;; Set the compose key to `CapsLock'
  cmp-seq-delay 5 ;; 5ms delay between each compose-key sequence press

  ;; Comment this if you want unhandled events not to be emitted
  fallthrough false

  ;; Set this to false to disable any command-execution in KMonad
  allow-cmd true
)

(defsrc
  esc  1    2    3    4    5    6    7    8    9    0    -    =    bspc
  tab  grv  q    w    e    r    t    y    u    i    o    p    ;    del
  caps \    a    s    d    f    g    h    j    k    l    '    ret
  lsft [    ]    z    x    c    v    b    n    m    ,    .    up   rsft
  lctl ralt lmet lalt ralt      spc            rctl ralt left down right
)

(deflayer qwerty
  esc  2    2    3    4    5    6    7    8    9    0    _    =    bspc
  tab  grv  q    w    e    r    t    y    u    i    o    p    ;    del
  caps \    a    s    d    f    g    h    j    k    l    '    ret
  lsft [    ]    z    x    c    v    b    n    m    ,    .    up   rsft
  lctl ralt lmet lalt ralt      spc            rctl ralt left down right
)
```

My keyboard looks like this (66 keys), I hope my configuration is correct:

![IMG-20230805-WA0001](https://github.com/kmonad/kmonad/assets/7107523/9247fc7e-fe06-46ef-85f1-557bced44342)

**Comment by Kabouik** [comment]
Since both `Ctrl` keys are wired together on the PCB, as well as both `Shift` keys, I also tried changing `rctl` into `lctl` and `rsft` into `lsft` in my config. I also removed the leftmost and rightmost `ralt` since evtest wouldn't see those keys in Droidian. However, the same issue persists.

There seem to be several issues related to that one since 2022, I wonder if I should try compiling an older commit?
